specs/language/basic.tex

```diff
@@ -1,5 +1,121 @@
+\item they are conversion-function-ids formed with the same type, or
+\item they are template-ids that refer to the same class or function.
+\end{itemize}
+
+\p \begin{note}
+  This section matches \gls{isoCPP} section \textbf{[basic]} except for the
+  exclusion of \texttt{goto} and \textit{literal operators}.
+\end{note}
+
+\Sec{Declarations and definitions}{Basic.Decl}
+
+\p A declaration (\ref{Decl}) may introduce one or more names into a translation
+unit or redeclare names introduced by previous declarations. If a declaration
+introduces names, it specifies the interpretation and attributes of these names.
+A declaration may also have effects such as:
+\begin{itemize}
+\item verifying a static assertion (\ref{Decl}),
+\item use of attributes (\ref{Decl}), and
+\item controlling template instantiation (\ref{Template.Inst}).
+\end{itemize}
+
+\p A declaration is a \textit{definition} unless:
+\begin{itemize}
+\item it declares a function without specifying the function's body
+(\ref{Decl.Function}),
+\item it is a parameter declaration in a function declaration that does not
+specify the function's body (\ref{Decl.Function}),
+\item it is a global or namespace member declaration without the \texttt{static}
+specifier\footnote{Global variable declarations are implicitly constant and
+external in HLSL.},
+\item it declares a static data member in a class definition,
+\item it is a class name declaration,
+\item it is a template parameter,
+\item it is a \texttt{typedef} declaration (\ref{Decl}),
+\item it is an \textit{alias-declaration} (\ref{Decl}),
+\item it is a \textit{using-declaration} (\ref{Decl}),
+\item it is a \textit{static\_assert-declaration} (\ref{Decl}),
+\item it is an \textit{empty-declaration} (\ref{Decl}),
+\item or a \textit{using-directive} (\ref{Decl}).
+\end{itemize}
+
+\p The two examples below is adapted from \gls{isoCPP} \textbf{[basic.def]}. All
+but one of the following are definitions:
+\begin{HLSL}
+int f(int x) { return x+1; } // defines f and x
+struct S {int a;int b;};     // defines S, S::a, and S::b
+struct X {                   // defines X
+  int x;                     // defines non-static member x
+  static int y;              // declares static data member y
+};
+int X::y = 1;                // defines X::y
+enum { up, down };           // defines up and down
+namespace N {                // defines N
+int d;                       // declares N::d
+static int i;                // defines N::i
+}
+\end{HLSL}
+
+\p All of the following are declarations:
+\begin{HLSL}
+int a;                       // declares a
+const int c;                 // declares c
+X anX;                       // declares anX
+int f(int);                  // declares f
+struct S;                    // declares S
+typedef int Int;             // declares Int
+using N::d;                  // declares d
+using Float = float;         // declares Float
+cbuffer CB {                 // does not declare CB
```

Comment:
Does `CB` have any significance after it appears in `cbuffer CB`?  It's not declared....so what is it?

Reply:
Yea... we need to write up an explanation of that elsewhere in the spec. It isn't a declared name in the HLSL translation unit, but all the declarations under it get grouped in reflection. It behaves a lot more like a grouping attribute than a declaration.

Reply:
Resolving conversation based on my comment above.